android/src/main/java/com/hypertrack/sdk/capacitor/CapacitorSerialization.kt

```diff
@@ -0,0 +1,138 @@
+package com.hypertrack.sdk.capacitor
+
+import android.util.Log
+import com.getcapacitor.JSArray
+import com.getcapacitor.JSObject
+import com.getcapacitor.PluginCall
+import com.hypertrack.sdk.capacitor.common.Result
+import com.hypertrack.sdk.capacitor.common.Failure
+import com.hypertrack.sdk.capacitor.common.Success
+import org.json.JSONArray
+import org.json.JSONObject
+
+@Suppress("UNCHECKED_CAST")
+internal fun <T> Result<T>.toPluginCall(call: PluginCall) {
+  when(this) {
+    is Success -> {
+      when(this.success) {
+        is Map<*, *> -> {
+            call.resolve((this.success as Map<String, Any>).toJSObject())
+        }
+        is Unit -> {
+            call.resolve()
+        }
+        else -> {
+            val exception = IllegalArgumentException(this.success.toString())
+            call.reject(exception.message, exception)
+        }
+      }
+    }
+    is Failure -> {
+        val exception = this.failure
+        if(exception is Exception) {
+            call.reject(exception.message, exception)
+        } else {
+            throw exception
+        }
+    }
+  }
+}
+
+@Suppress("UNCHECKED_CAST")
+internal fun List<Any>.toJSArray(): JSArray {
+    return JSArray().also { writableArray ->
+        forEach {
+            when (it) {
+                is String -> {
+                    writableArray.put(it)
+                }
+                is Map<*, *> -> {
+                    writableArray.put((it as Map<String, Any>).toJSObject())
+                }
+                else -> {
+                    throw Exception(IllegalArgumentException(it.javaClass.toString()))
+                }
+            }
+        }
+    }
+}
+
+@Suppress("UNCHECKED_CAST")
+internal fun Map<String, Any?>.toJSObject(): JSObject {
+    val map = this
+    return JSObject().apply {
+        map.forEach { entry ->
```

Comment:
Related to the comment above, please use deconstruction operator:

`forEach { (key, value) -> `

Reply:
+

Reply:
It doesnt look like it's fixed

Reply:
Now it's fixed (there was Github bug)